plaid/src/functions/api.rs

```diff
@@ -130,6 +130,95 @@ macro_rules! impl_new_function_with_error_buffer {
     }
 }
 
+/// Macro to implement a function in a specific API's submodule.
+///
+/// This macro generates two functions:
+/// - A private implementation function (`_impl`) that handles the actual logic:
+///   - Accessing and validating memory from the guest
+///   - Checking that the API is configured.
+///   - Running the function + returning the result and handling errors
+/// - A public wrapper function that calls the implementation function and returns the result as an integer.
+///
+/// # Parameters
+/// - `$api`: The name of the API (e.g., `aws`).
+/// - `$sub_module`: The name of the submodule within the API (e.g., `kms`).
+/// - `$function_name`: The name of the function to be implemented (e.g., `put_object`, `encrypt`).
+///
+/// # Error Handling
+/// The generated implementation function returns `FunctionErrors` in case of failures, which are then
+/// converted to int error codes by the wrapper function. These errors include:
+/// - `FunctionErrors::InternalApiError`: For internal API-related errors.
+/// - `FunctionErrors::ApiNotConfigured`: If the API is not configured.
+/// - `FunctionErrors::ReturnBufferTooSmall`: If the provided return buffer is too small to hold the result.
+macro_rules! impl_new_sub_module_function_with_error_buffer {
```

Comment:
I think to crush the two very similar macros together you need to modify this so it expects the submodule to have the trailing `.`.

Then you have an interim function with this name `impl_new_sub_module_function_with_error_buffer` which add the final `.` to the submodule before calling this.

And the existing one can call it directly because there is no submodule. Do you think that may work?

Reply:
Worth trying!